core/pom.xml

```diff
@@ -215,6 +215,11 @@
       <version>3.2.10</version>
     </dependency>
     <dependency>
+      <groupId>com.fasterxml.jackson.module</groupId>
+      <artifactId>jackson-module-scala_2.10</artifactId>
+      <version>2.3.1</version>
```

Comment:
what i was asking @pwendell/@andrewor14 was the probability this breaks user applications (or user applications break Spark) due to version conflicts for both Jackson & the Scala module.